### DISCUSSION TITLE
What are the best X86 and X64 memory locations for booting an OS from the Programmable ROM?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#101126] [Programming Help]
### Here is how the OS will work:

* Use org to set the memory addresses for where the bootloader and kernel should be stored.

* Allocate bytes.

* Jump to those memory locations.

* Kernel will be stored into RAM.

* Bootloader will be stored into ROM.

##### RAM:

* Physical Address Range: ```0x00000000```-  ```0xFFFFFFFF (4GB)```
* Usable Memory Range (approximate): ```0x00000000``` -``` 0x3FFFFFFF ```(1GB) due to reserved addresses for system devices and BIOS

### PROM:

* Programmable ROM Address:```0xFFFFFFF0```
* Jump to the RAM to load the kernel

```assembly
bits 32
org 0xFFFFFFF0
section .data 
 bootloader: times 52 db 0 
section .text
 global _start
_start:
 ; Load the bootloader
  mov ecx, bootloader
  mov esi, 52 
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by DarrenChochian** [comment]


**Reply by ghost** [reply]
I am looking for programmers to help me make this, but I don't have any money. If you are willing to can you please help me? 

It will be coded in NASM X86.
